feat: added dahsboard and template

cobcan opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines in the [CONTRIBUTING.md](https://github.com/falcosecurity/.github/blob/master/CONTRIBUTING.md) of the main Falco repository.
2. Please label this pull request according to what type of issue you are addressing.
3. If the PR is unfinished while opening it specify a wip in the title before the actual title, for example, "wip: my awesome feature"
-->

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> If this PR will release a new chart version please make sure to also uncomment the following line:

/kind chart-release

<!--
Please remove the leading whitespace before the `/kind <>` you uncommented.
-->

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area falco-chart

> /area falco-exporter-chart

/area falcosidekick-chart

> /area falco-talon

> /area event-generator-chart

> /area k8s-metacollector

<!--
Please remove the leading whitespace before the `/area <>` you uncommented.
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
If PR is `kind/failing-tests` or `kind/flaky-test`, please post the related issues/tests in a comment and do not use `Fixes`.
-->

Fixes #

**Special notes for your reviewer**:

**Checklist**
<!--
Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.
-->
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] CHANGELOG.md updated

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [ab8a88e](https://github.com/falcosecurity/charts/commits/ab8a88e7bcb65f79d859969eb59662a58ca0a58d) feat: added dahsboard and template

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/charts/pull/795#" title="Author self-approved">cobcan</a>*
**Once this PR has been reviewed and has the lgtm label**, please assign [cpanato](https://github.com/cpanato) for approval. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Fcharts).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/charts/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["cpanato"]} -->

cobcan commented:
@alacuku @Issif Sorry, I closed the other PR accidentally #792

leogr commented:
@cobcan Thank you :pray: 

I just noticed this, may you fix it please?

![image](https://github.com/user-attachments/assets/8eaad4ae-5471-4fd3-9959-6dd7c205f4c6)

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [ea7e401](https://github.com/falcosecurity/charts/commits/ea7e401d132366fe05cd9f6088e0d05d1281719e) fix identation

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [ea7e401](https://github.com/falcosecurity/charts/commits/ea7e401d132366fe05cd9f6088e0d05d1281719e) fix identation
- [f1e087e](https://github.com/falcosecurity/charts/commits/f1e087ef46505ba8a6d3aedff33bae853e9b9d48) change default value

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [ea7e401](https://github.com/falcosecurity/charts/commits/ea7e401d132366fe05cd9f6088e0d05d1281719e) fix identation
- [f1e087e](https://github.com/falcosecurity/charts/commits/f1e087ef46505ba8a6d3aedff33bae853e9b9d48) change default value

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
Adding label `do-not-merge/contains-merge-commits` because PR contains merge commits, which are not allowed in this repository.
Use `git rebase` to reapply your commits on top of the target branch. Detailed instructions for doing so can be found [here](https://git.k8s.io/community/contributors/guide/github-workflow.md#4-keep-your-branch-in-sync).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository.
</details>

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [ea7e401](https://github.com/falcosecurity/charts/commits/ea7e401d132366fe05cd9f6088e0d05d1281719e) fix identation
- [f1e087e](https://github.com/falcosecurity/charts/commits/f1e087ef46505ba8a6d3aedff33bae853e9b9d48) change default value
- [ab8381f](https://github.com/falcosecurity/charts/commits/ab8381fdac9c5de69068c8fec9b64a61298703b7) fix identation

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [ea7e401](https://github.com/falcosecurity/charts/commits/ea7e401d132366fe05cd9f6088e0d05d1281719e) fix identation
- [f1e087e](https://github.com/falcosecurity/charts/commits/f1e087ef46505ba8a6d3aedff33bae853e9b9d48) change default value
- [ab8381f](https://github.com/falcosecurity/charts/commits/ab8381fdac9c5de69068c8fec9b64a61298703b7) fix identation

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

leogr commented:
Hey @cobcan 

I noticed the rebase did not work. I will fix it for you, please hold on.

cobcan commented:
@leogr 

okay thank you

leogr commented:
@cobcan we should be ok now.

summoning @Issif for the last check and approval :pray:

Issif commented on charts/falcosidekick/values.yaml:
```diff
@@ -722,6 +722,19 @@ config:
     # -- minimum priority of event to use this output, order is `emergency\|alert\|critical\|error\|warning\|notice\|informational\|debug or ""`
     minimumpriority: ""
+    # -- dashboards contains configuration for grafana dashboards.
+    dashboards:
+      falcosidekickLoki:
+        # -- enabled specifies whether this dashboard should be deployed.
+        enabled: false
+        # --configmaps to be deployed that contain a grafana dashboard.
+        configMap:
+          # -- name specifies the name for the configmap.
+          name: falcosidekick-loki-dashboard-grafana
+          # -- namespace specifies the namespace for the configmap.
+          namespace: ""
+          # -- folder where the dashboard is stored by grafana.
+          folder: ""
```
there's a mismatch here, this output `grafana` creates annotations, this block should in the `Loki` section.

something like:

```yaml
  loki:
    # -- Loki <http://host:port>, if not `empty`, Loki is *enabled*
    hostport: ""
    # -- user for Grafana Logs
    user: ""
    # -- API Key for Grafana Logs
    apikey: ""
    # -- Loki endpoint URL path, more info: <https://grafana.com/docs/loki/latest/api/#post-apiprompush>
    endpoint: "/loki/api/v1/push"
    # -- Loki tenant, if not `empty`, Loki tenant is *enabled*
    tenant: ""
    # -- comma separated list of fields to use as labels additionally to rule, source, priority, tags and custom_fields
    extralabels: ""
    # -- a list of comma separated custom headers to add, syntax is "key:value,key:value"
    customheaders: ""
    # -- minimum priority of event to use this output, order is `emergency\|alert\|critical\|error\|warning\|notice\|informational\|debug or ""`
    minimumpriority: ""
    # -- if true, checkcert flag will be ignored (server cert will always be checked)
    mutualtls: false
    # -- check if ssl certificate of the output is valid
    checkcert: true
    # -- dashboard for Grafana
    grafanaDashboard:
      # -- enabled specifies whether this dashboard should be deployed.
      enabled: false
      # --configmaps to be deployed that contain a grafana dashboard.
      configMap:
        # -- name specifies the name for the configmap.
        name: falcosidekick-loki-dashboard-grafana
        # -- namespace specifies the namespace for the configmap.
        namespace: ""
        # -- folder where the dashboard is stored by grafana.
        folder: ""
```

cobcan replied:
the change suggested is done, but one question. What do you mean that "grafana create annotations"??

Issif replied:
the section `grafana` is to configure the output of falcosidekick which creates annotations for the graphs with the falco events: https://grafana.com/docs/grafana/latest/dashboards/build-dashboards/annotate-visualizations/

Issif reviewed (commented):
I see some points to improve:
- this dashboard is for Loki but also relies on Prometheus to fill the variables for templating, can you have just Loki as dependency?
- the dashboard is named `Falco Pods`, `Falco logs` should be more relevant
- you use the label `cluster` for the template, are you sure it's always available, it's not preset either by Falco either by Falcosidekick
- some panels don't have any name, 1 is called `Panel Title`
- your count by for the table should also count by namespace to be exhaustive

leogr commented:
> @leogr
> 
> thanks for doing your git-magic 😄

np! If you plan to add more commits, I recommend you remove your local repo, and git clone it again (so you will be sure to be aligned with your fork)

cobcan commented:
> I see some points to improve:
> 
> * this dashboard is for Loki but also relies on Prometheus to fill the variables for templating, can you have just Loki as dependency?
> * the dashboard is named `Falco Pods`, `Falco logs` should be more relevant
> * you use the label `cluster` for the template, are you sure it's always available, it's not preset either by Falco either by Falcosidekick
> * some panels don't have any name, 1 is called `Panel Title`
> * your count by for the table should also count by namespace to be exhaustive



1. I've change it to get the namespaces from loki
2. It's changed to Falco logs
3. "cluster" was a variable from Prometheus to make de dashboard multi-cluster, but since Prometheus is removed is no longer needed
4. I've added the label 'k8s_ns' to the table to check where namespace is each pod.
5. Extra: I've modified the rates panels, the ones at the bottom of the dashboard, to be also filtered by namespace

Issif commented:
/lgtm

poiana commented:
LGTM label has been added.  <details>Git tree hash: 297cb24ab3d5012275eac50e4e0fb55eb4754132</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/charts/pull/795#" title="Author self-approved">cobcan</a>*, *<a href="https://github.com/falcosecurity/charts/pull/795#issuecomment-2540981097" title="LGTM">Issif</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Fcharts).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/charts/blob/master/OWNERS)~~ [Issif]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/charts/pull/795#" title="Author self-approved">cobcan</a>*, *<a href="https://github.com/falcosecurity/charts/pull/795#issuecomment-2540981097" title="LGTM">Issif</a>*, *<a href="https://github.com/falcosecurity/charts/pull/795#pullrequestreview-2505755866" title="Approved">leogr</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Fcharts).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/charts/blob/master/OWNERS)~~ [Issif,leogr]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
